### ISSUE TITLE
Bad error message at `by` block

### ISSUE THREAD

**Issue by leodemoura** [leanprover/lean4#1349]
```lean
syntax "foo " ident : tactic

example : True := by -- Error here: invalid 'by' tactic, expected type has not been provided
  foo x + 1
```
We got the weird error message above because Lean parsed the example above as
```lean
example : True := (by foo x) + 1
```

**Comment by gebner** [comment]
See also #1359.

**Comment by leodemoura** [comment]
The fix for issue #1359 improved the error message here. For the issue above, we now get
```lean
error: type mismatch
  ?m.74 + 1
has type
  Nat : Type
but is expected to have type
  True : Prop
```
If we write it as
```lean
syntax "foo " ident : tactic

example : Nat := by 
  foo x + 1
```
We get an error at `foo x` saying the tactic has not been implemented.
So, I am closing this issue.
